codalab/model/bundle_model.py

```diff
@@ -986,11 +986,15 @@ def save_bundle(self, bundle):
             self.do_multirow_insert(connection, cl_bundle_metadata, metadata_values)
             bundle.id = result.lastrowid
 
-    def update_bundle(self, bundle, update, connection=None):
+    def update_bundle(self, bundle, update={}, connection=None, delete_keys=[]):
         """
-        Update a bundle's columns and metadata in the database and in memory.
-        The update is done as a diff: columns that do not appear in the update dict
-        and metadata keys that do not appear in the metadata sub-dict are unaffected.
+        For each (key, value) pair in update dictionary, add or update metadata pair (key, value).
```

Comment:
Actually, just realized that `update` contains "metadata".  Make this more precise.  Then `delete_metadata_keys` is more precise.

Reply:
It's updated!